### DISCUSSION TITLE
为无边框应用添加窗口圆角

### DISCUSSION THREAD

**Discussion by Hibanaw** [linuxdeepin/developer-center#3183] [Features Request & Ideas | 特性请求 & 头脑风暴]
为无边框应用添加与 dtk 应用一致的窗口圆角，类似 Windows 11 的效果

**Comment by BLumia** [comment]
虽然可以通过 KWin 脚本做到这种效果，但无边框（Client-Side Decoration）应用的呈现形式本身就应该由应用自身控制，而不是受系统管理呀，这样说不定会影响一些应用自身的行为。

**Reply by Hibanaw** [reply]
因为我发现一些在其他 de 中无阴影的应用（比如 electron 应用和 deepin-wine wechat）在 dde 内都有阴影，不知道是不是 dde 进行了一些匹配
给这些主动添加阴影的窗口加上圆角应该是合理的？

**Reply by BLumia** [reply]
应用可以设置一些窗口属性来设置是否使用阴影的，在有对应的属性的时候，窗管就可以为其绘制圆角/阴影了，DDE 下有阴影的话 KDE 下应该也会有，可以试试看。

**Reply by Hibanaw** [reply]
> 应用可以设置一些窗口属性来设置是否使用阴影的，在有对应的属性的时候，窗管就可以为其绘制属性了，DDE 下有阴影的话 KDE 下应该也会有，可以试试看。

现在还没来得及尝试 kde 下的情况，我印象中之前是没有的

主要是大部分 electron 应用在大圆角基调的 DDE 下却是直角，看起来还是很割裂的
这些应用在 Windows 下倒是都有系统绘制的窗口圆角

另外，系统边框的圆角半径跟 dtk 圆角半径也不同，这个可以统一一下嘛

**Reply by zccrs** [reply]
主要原因是因为我们默认的圆角太大了，的确能自动为有阴影的窗口统一加圆角，不过这个大小不能跟 DTK 的程序保持一致，必须要限制最大半径小于 8，不然可能会裁掉一部分有用的内容。这类的设置后边可以放到一个独立的高级设置工具中，现在正在规划这样的工具。CC @meiyixiang 

在高级工具中提供这些设置就不用考虑这么多了。

**Comment by meiyixiang** [comment]
谢谢你的建议，我们会考虑将他放到我们的高级设置管理工具中
